File: python/post_example.py
import logging
import hashlib
import json
import jwt
import requests

ACCESS_KEY = 'xxxxx'
SECRET_KEY = 'xxxxx'

logger = logging.getLogger(__name__)

path = "/companies"
body = json.dumps({'companies': [{'name': 'XYZ Company', 'jurisdictions': ['Maine'], 'entity_type': 'Limited Liability Company'}]})

headers = {"access_key": ACCESS_KEY}
payload = {
    "path": path,
    "content": hashlib.sha256(body.encode('ascii')).hexdigest()
}

token = jwt.encode(payload, SECRET_KEY, algorithm='HS256', headers=headers).decode('utf-8')

try:
    url = "https://api.corporatetools.com/companies"
    headers = {"Authorization": 'Bearer %s'%(token), "Content-Type": "json"}
    r = requests.post(url, headers=headers, data=body)

    print(r.json())
except Exception as e:
    logger.error(msg="{e}".format(e=e))
    raise e

File: javascript/post_example.js
exports.post = {
	postCompanies: function(apiUrl, jwt, CryptoJS, request, keys) {
		let body = JSON.stringify(
			{ companies: [
				{ 
					name: 'Example Company', 
					entity_type: 'Limited Liability Company', 
					jurisdictions: ['Maine', 'Washington']
				}]
			}
		); 

		let payload = {
		  path: '/companies',
		  content: CryptoJS.SHA256(body).toString(CryptoJS.enc.Hex)
		}; 

		let header = { access_key: keys.access_key }; 

		let token = jwt.encode(payload, keys.secret_key, 'HS256', { header: header });

		let url = apiUrl + '/companies'

		request.open('POST', url, true);
		request.setRequestHeader('Authorization', 'Bearer ' + token);
		request.setRequestHeader("Content-Type", "application/json");

		request.onload = function () {
		    if (request.readyState === 4 && request.status === 200) {
		        let response = JSON.parse(request.responseText);
		        console.log(response);
		    } else {
		    	console.log("Error", JSON.parse(request.responseText)); 
		    }
		};

		request.send(body);
	}
}